Packages/com.unity.inputsystem/Tests/InputSystem/Monobehaviors/TrackedPoseDriverTests.cs

```diff
@@ -0,0 +1,155 @@
+            }
+        }
+
+        static Vector3 testpos = new Vector3(1.0f, 2.0f, 3.0f);
+        static Quaternion testrot = new Quaternion(0.09853293f, 0.09853293f, 0.09853293f, 0.9853293f);
+
+
+        internal class TestPoseProvider : BasePoseProvider
+        {
+            public override bool TryGetPoseFromProvider(out Pose output)
+            {
+                Pose tmp = new Pose();                
+                tmp.position = testpos;
+                tmp.rotation = testrot;
+                output = tmp;
+                return true;
+            }
+        }
+
+        internal static TestTrackedPoseDriverWrapper CreateGameObjectWithTPD()
+        {
+            GameObject go = new GameObject();
+            TestTrackedPoseDriverWrapper tpd = go.AddComponent<TestTrackedPoseDriverWrapper>();
+            return tpd;
+        }
+
+        internal static BasePoseProvider CreatePoseProviderOnTPD(TestTrackedPoseDriverWrapper tpd)
+        {
+            TestPoseProvider tpp = tpd.gameObject.AddComponent<TestPoseProvider>();
+            tpd.poseProviderComponent = tpp;
+            return tpp;
+        }
+
+
+        [TearDown]
+        public void TearDown()
+        {
+
+        }
+
+        [Test]
+        public void TPDBindingTest()
+        {
+            var baseMap = new InputActionMap("Base");
+            var action1 = baseMap.AddAction("action1", binding: "<Gamepad>/buttonSouth");
+            var action2 = baseMap.AddAction("action2", binding: "<Gamepad>/buttonNorth");
+
+            TestTrackedPoseDriverWrapper tpd = CreateGameObjectWithTPD();
+            tpd.positionAction = action1;
+            Assert.That(tpd.positionAction.bindings[0].path, Is.EqualTo("<Gamepad>/buttonSouth"));
+            tpd.positionAction = action2;
+            Assert.That(tpd.positionAction.bindings[0].path, Is.EqualTo("<Gamepad>/buttonNorth"));
+
+            
+            tpd.rotationAction = action1;
+            Assert.That(tpd.rotationAction.bindings[0].path, Is.EqualTo("<Gamepad>/buttonSouth"));
+            tpd.rotationAction = action2;
+            Assert.That(tpd.rotationAction.bindings[0].path, Is.EqualTo("<Gamepad>/buttonNorth"));        
+        }
+
+
+        [Test]
+        public void TPDPoseProviderTest()
+        {
+            TestTrackedPoseDriverWrapper tpd = CreateGameObjectWithTPD();
+            BasePoseProvider pp = CreatePoseProviderOnTPD(tpd);
+
+            Assert.That(tpd.poseProviderComponent, Is.EqualTo(pp));
+
+            tpd.FakeUpdate();
+            Assert.That(tpd.gameObject.transform.position, Is.EqualTo(testpos));
+            Assert.That(tpd.gameObject.transform.rotation.Equals(testrot));
+
+        }
+
+        public void Reset(GameObject go)
+        {
+            go.transform.position = new Vector3(0.0f, 0.0f, 0.0f);
+            go.transform.rotation = new Quaternion(0.0f, 0.0f, 0.0f, 0.0f);
+        }
+
+        [Test]
+        public void TPDUpdateOptionTest()

```

Comment:
No `XXXTest()` tests. All tests should express the functionality that is being established or the behavior that is being verified. `XXXTest` is a bad pattern. You can't know what the test is really about until you've actually read it.

For guidance, take a look at the other tests. Good names are things like `CanUpdateStateOfDeviceWithEvent`. Out of a good name usually follows better content, too. `XXXTest()` tend to lack focus by never establishing what exactly is being verified.

Reply:
On it.